js/src/api/contract/contract.js

```diff
@@ -209,9 +221,93 @@ export default class Contract {
     return `0x${data || ''}${call || ''}`;
   }
 
-  _encodeOptions (func, options, values) {
+  /**
+   * Check whether the given address could be
+   * a Wallet. The result is cached in order not
+   * to make unnecessary calls on non wallet accounts
+   */
+  _isWallet (address) {
```

Comment:
Not 100% sure this is the correct place for this functionality. Can we walk through today and discuss?